internal/processors/participants.go

```diff
@@ -0,0 +1,120 @@
+			participant = participantsForLedgerEntry(c.MustCreated())
+		case xdr.LedgerEntryChangeTypeLedgerEntryRemoved:
+			participant = participantsForLedgerKey(c.MustRemoved())
+		case xdr.LedgerEntryChangeTypeLedgerEntryUpdated:
+			participant = participantsForLedgerEntry(c.MustUpdated())
+		case xdr.LedgerEntryChangeTypeLedgerEntryState:
+			participant = participantsForLedgerEntry(c.MustState())
+		default:
+			return nil, fmt.Errorf("unknown ledger entrychange type %d", c.Type)
+		}
+
+		if participant != nil {
+			participants = append(participants, *participant)
+		}
+	}
+
+	return participants, nil
+}
+
+func participantsForLedgerEntry(le xdr.LedgerEntry) *xdr.AccountId {
+	if le.Data.Type != xdr.LedgerEntryTypeAccount {
+		return nil
+	}
+	aid := le.Data.MustAccount().AccountId
+	return &aid
+}
+
+func participantsForLedgerKey(lk xdr.LedgerKey) *xdr.AccountId {
+	if lk.Type != xdr.LedgerEntryTypeAccount {
+		return nil
+	}
+	aid := lk.MustAccount().AccountId
+	return &aid
+}
+
+// participantsForMeta identifies the participants involved in a transaction meta by looping through the operations' changes.
+func participantsForMeta(meta xdr.TransactionMeta) ([]xdr.AccountId, error) {
+	var participants []xdr.AccountId
+	if meta.Operations == nil {
+		return participants, nil
+	}
+
+	for _, op := range *meta.Operations {
+		var accounts []xdr.AccountId
+		accounts, err := participantsForChanges(op.Changes)
+		if err != nil {
+			return nil, fmt.Errorf("identifying participants for changes: %w", err)
+		}
+
+		participants = append(participants, accounts...)
+	}
+
+	return participants, nil
+}
+
+func (p *ParticipantsProcessor) GetTransactionParticipants(transaction ingest.LedgerTransaction) (set.Set[string], error) {
+	// 1. Get direct participants involved in the transaction
+	participants := []xdr.AccountId{
+		transaction.Envelope.SourceAccount().ToAccountId(), // in case of a fee bump, this is the innerTx source account
+	}
+	if transaction.Envelope.IsFeeBump() {
+		participants = append(participants, transaction.Envelope.FeeBumpAccount().ToAccountId())
+	}
+
+	feeParticipants, err := participantsForChanges(transaction.FeeChanges)
+	if err != nil {
+		return nil, fmt.Errorf("identifying participants for changes: %w", err)
+	}
+	participants = append(participants, feeParticipants...)
+
+	// 1.1. Get participants involved in the transaction meta (if successful)
+	if transaction.Result.Successful() {
+		metaParticipants, metaErr := participantsForMeta(transaction.UnsafeMeta)
+		if metaErr != nil {
+			return nil, fmt.Errorf("identifying participants for meta: %w", metaErr)
+		}
+		participants = append(participants, metaParticipants...)
+	}
+
+	// 2. Push transaction and participants to data bundle
+	participantsSet := set.NewSet[string]()
+	for _, xdrParticipant := range participants {
+		participantsSet.Add(xdrParticipant.Address())
+	}
+
+	return participantsSet, nil
+}
+
+type OperationParticipants struct {
```

Comment:
I am assuming this is to be added in another PR?

Reply:
https://github.com/stellar/wallet-backend/pull/219/commits/5012c5353e18fae75c3238607de0dc8cd1a83b43